### PR TITLE
Chainparams: decouple anyonecanspend are mine from regtest

### PR DESCRIPTION
**Pull request by jtimon**
Spending op_true outputs has more to do with the wallet than with the chain. Perhaps this options should be per wallet in the multiwallet future?

Alternatively we can make it a field in CChainParams, but ```Params().NetworkIDString() == CHAINPARAMS_REGTEST``` defeats the whole point of having chainparams IMO.

### CODE REVIEW

File: src/script/ismine.h [hunk 1]
```diff
@@ -13,6 +13,8 @@
 class CKeyStore;
 class CScript;
 
+static const bool DEFAULT_ANYONECANSPEND_AREMINE = true;
```
Comment (instagibbs) [inline src/script/ismine.h]: imo this should be false by default. It's a hack for regtest, for the most part.

File: src/script/ismine.h [hunk 2]
```diff
@@ -28,6 +30,9 @@ enum isminetype
 /** used for bitflags of isminetype */
 typedef uint8_t isminefilter;
 
+// TODO support for multiwallet ?
+extern bool g_anyonecanspend_aremine;
```
Comment (instagibbs) [inline src/script/ismine.h]: Not sure this needs multiwallet support. It's for testing only.

**Comment by instagibbs** [comment]
would be easy to add a basic test; please do

File: src/script/ismine.h (hunk 2 above)
Comment (jtimon) [reply src/script/ismine.h]: I'll remove the comment, sorry.

File: src/script/ismine.h (hunk 1 above)
Comment (jtimon) [reply src/script/ismine.h]: Mhmm, my thought was that custom/regtest is the only chain that we support directly...perhaps it should be a field in CChainParams instead?
The important part is removing ```Params().NetworkIDString() == CHAINPARAMS_REGTEST```

**Comment by jtimon** [comment]
Updated making a new field in CChainParams instead of an independent global.
Now it requires the custom chain to be configurable, so the test is in the last commit of https://github.com/ElementsProject/elements/pull/274

**Comment by instagibbs** [comment]
utACK

**Pull request merged**